### DISCUSSION TITLE
Does larastan support support dynamic where's on base `Query\Builder`, or only on `Eloquent\Builder`?

### DISCUSSION THREAD

**Discussion by claytonrcarter** [larastan/larastan#1826] [Q&A]
Hi! I'm using current `larastan` (v2.8.1) and noticed that dynamic wheres aren't recognized when using the `DB` facade. Am I doing something wrong (most likely!) or are they only supported on `Eloquent\Builder`?

```php
// this code works, and phpstan is OK with it
Order::whereNumber(123);

// this code also works, but phpstan doesn't like it, with error
// Call to an undefined method Illuminate\Database\Query\Builder::whereNumber()
DB::table('orders')->whereNumber(123);
```

Is this intentional (maybe because we can't know if there is a `number` column in the `orders` table?), or a bug, or have I just missed something?

**Comment by canvural** [comment]
Hi,

Yes it is only supported in Eloquent builder. Because then we know the model. As the DB facade can work with any arbitrary table or connection, it is highly unlikely we will support this.
